### DISCUSSION TITLE
Error with Nitrogen Valence - Explicit valence for atom # 1 N, 4, is greater than permitted

### DISCUSSION THREAD

**Discussion by rsinduja** [rdkit/rdkit#4829] [Q&A]
Hello,

I am facing an issue with generating fingerprints for the structures with Nitrogen bonds for example,

[H][N]([H])([H])[Pt]1(OC(=O)C2(CCC2)C(=O)O1)[N]([H])([H])[H]

Explicit valence for atom # 1 N, 4, is greater than permitted

I tried turning off the sanitization on molecule construction and then manually sanitizing with the valence check turned off.

    def __init__(self, smiles):
        self.mols = [Chem.MolFromSmiles(i, sanitize=False) for i in smiles]
        self.mols.UpdatePropertyCache(strict=False)
  Chem.SanitizeMol(self.mols,Chem.SANITIZE_SYMMRINGS|Chem.SANITIZE_SETCONJUGATION|Chem.SANITIZE_SETHYBRIDIZATION)

But, again, the RDKit's valence rules are not allowing me to generate the fingerprint.

Kindly help me with this issue.

Thank
![Capture](https://user-images.githubusercontent.com/75007432/147577633-904c032f-b41a-42c7-8cf5-32f33f1bc8b8.PNG)
s

**Comment by greglandrum** [comment]
The SMILES you provide above doesn't raise an error:
```
In [2]: m = Chem.MolFromSmiles('[H]N([H])[Pt]1(OC(=O)C2(CCC2)C(=O)O1)N([H])[H]')

In [3]: m.GetNumAtoms()
Out[3]: 13
```
and the code you show in the example isn't going to work even with a SMILES which doesn't include a valence error

If you want us to be able to help or answer questions, you need to share the code you actually used, the exact input, and the error messsage you saw.

**Reply by rsinduja** [reply]
When I copy-pasted the smiles I suppose the braces were automatically deleted. Below mentioned is the original code and the SMILES I used for fingerprint generation.

from rdkit import Chem
from rdkit.Chem import AllChem, Draw
from molvs import standardize_smiles
smiles = standardize_smiles("[H][N]([H])([H])[Pt]1(OC(=O)C2(CCC2)C(=O)O1)[N]([H])([H])[H]")
mol = Chem.MolFromSmiles(smiles)
bit_morgan1 = {}
fp = AllChem.GetMorganFingerprintAsBitVect(mol, radius=3, nBits=2048, bitInfo=bit_morgan1)
bits = list(fp)

**Reply by greglandrum** [reply]
The SMILES you are trying to parse there comes from `molvs.standardize_smiles()`, so you would need to take a look at what that is in order to understand where the bad valence problem is coming from.

For what it's worth: there's an RDKit port of molvs that you can use, and that has no problem with this molecule:
```
In [2]: from rdkit.Chem.MolStandardize import rdMolStandardize

In [3]: rdMolStandardize.StandardizeSmiles("[H]N([H])[Pt]1(OC(=O)C2(CCC2)C(=O)O1)N([H])[H]")
[09:05:35] Initializing MetalDisconnector
[09:05:35] Running MetalDisconnector
[09:05:35] Removed covalent bond between Pt and N
[09:05:35] Removed covalent bond between Pt and O
[09:05:35] Removed covalent bond between Pt and N
[09:05:35] Removed covalent bond between Pt and O
[09:05:35] Initializing Normalizer
[09:05:35] Running Normalizer
Out[3]: 'O=C([O-])C1(C(=O)[O-])CCC1.[NH2-].[NH2-].[Pt+4]'
```

**Reply by rsinduja** [reply]
Thank you, this helped !
